tests/ExcelProvider.Tests/ExcelProvider.Tests.fs

```diff
@@ -380,3 +380,100 @@ let ``HashHeader false with header removed``() =
     let expectedTime = new DateTime(1899, 12, 31, 8, 0, 0)
     row.Column5 |> should equal expectedTime
     row.Column6 |> should equal 100.0M
+
+module AutomaticCoercion = 
+  type MixedDataTypes = ExcelFile<"MixedDataTypes.xlsx", "ValidDataTypes">
+
+  type DateTimeFromTextNumeric = ExcelFile<"MixedDataTypes.xlsx", "ValidDataTypes", Range="H2:H3", HasHeaders=false>
+  type DateTimeFromTextString = ExcelFile<"MixedDataTypes.xlsx", "ValidDataTypes", Range="H4:H5", HasHeaders=false>
+  type DateTimeFromNumeric = ExcelFile<"MixedDataTypes.xlsx", "ValidDataTypes", Range="H6:H7", HasHeaders=false>
+  type DateTimeFromGeneralNumeric = ExcelFile<"MixedDataTypes.xlsx", "ValidDataTypes", Range="H8:H9", HasHeaders=false>
+
+  type DoubleCoercionErrorFromNumberWords = ExcelFile<"MixedDataTypes.xlsx", "InvalidDataTypes", Range="B2:B3", HasHeaders=false>
+  type DoubleCoercionErrorFromString = ExcelFile<"MixedDataTypes.xlsx", "InvalidDataTypes", Range="B4:B5", HasHeaders=false>
+  type DoubleCoercionErrorFromCurrencyNumber = ExcelFile<"MixedDataTypes.xlsx", "InvalidDataTypes", Range="B6:B7", HasHeaders=false>
+
+  type DateCoercionErrorFromDateWords = ExcelFile<"MixedDataTypes.xlsx", "InvalidDataTypes", Range="H2:H3", HasHeaders=false>
+  type DateCoercionErrorFromString = ExcelFile<"MixedDataTypes.xlsx", "InvalidDataTypes", Range="H4:H5", HasHeaders=false>
+  type DateCoercionErrorFromNegativeNumber = ExcelFile<"MixedDataTypes.xlsx", "InvalidDataTypes", Range="H6:H7", HasHeaders=false>
```

Comment:
These are just sniping particular cells where the top cell is the expected type, and the lower cell is the type under test.